Contributors guide (#11)
ObieSource/obiesource.github.io: README.md

## Step 1: Set up Git:

Follow this guide for help setting it up:
https://docs.github.com/en/github-ae@latest/get-started/quickstart/set-up-git
This is the hardest part, and we recommend doing so in club so that we can answer questions while you do this.

## Step 2: Fork the repository:

A fork is a personal copy of someone else's repository. You can make whatever changes you want without messing up anyone else's work.
At the top of this page is a "fork" button. Click this button to make a personal copy of the obiesource website repository.

## Step 3: Clone the repository locally:

1. Navigate to your forked repository
	- If github doesn't automatically take you there, you can click your avatar in the top right corner > My repositories > obiesource.github.io
2. Click Code, and copy the link
3. Open a terminal on your computer (e.g. Command Prompt for Windows)
4. Navigate to where you want to store the repository locally
5. Enter the following command in your terminal:
	> git clone {the-link-you-copied}
6. Make a new branch (optional):
	- enter the following commands:
	> git branch new-member  

> 	> git checkout new-member

Reviewer: It's possible to do these both in one step. This way is less readable though, so feel free to take or leave this suggestion.

```suggestion
	> git checkout -b new-member
```

Author: Oooh thank you. Looks like I need to update how I make branches :)

## Step 4: Make changes:

1. Navigate to .../obiesource.github.io/members
2. Make a copy of template.json
3. Rename the copy {your-last-name}.json
4. Edit the document with your own personal information. All fields are optional (just delete them to remove them from the page)
5. Edit memberslist.json, and add yourself to the members field

> 	- the addition should look like this: {"name": "Your Name", "href": "/member.html?{your-last-name}"}

Reviewer: ```suggestion
	- the addition should look like this: `{"name": "Your Name", "href": "/member.html?{your-last-name}"}`
```

Adding backticks formats it as code in Markdown.

Author: Nice catch. Didn't know you could do this in md

	- correct spelling and quotation and comma placement are very imporant

## Step 5: Test your changes:

1. In your terminal, navigate to .../obiesource.github.io/
2. Run the following command:

> 	> python3 -m http.server --bind 127.0.0.1 8000

Reviewer: Should we say something about installing Python?

Reviewer 2: That might be a good thing to add at the beginning, like a list of dependencies. However, like the `git` CLI application, python3 is likely pre-installed on all computers, as far as i know.

Author: Hmm what do you think about:
1. In your terminal, navigate to .../obieousrce.github.io/
2. If you haven't already, install Python
   - run > python --version
   - if you receive an error that says "Python was not found," run > python
3. Run the following command:
     > python3 -m whatever

?

Reviewer: > That might be a good thing to add at the beginning, like a list of dependencies. However, like the `git` CLI application, python3 is likely pre-installed on all computers, as far as i know.

I think git and Python are preinstalled on Macs (or at least once you've installed the Xcode Developer Tools). But I didn't think they were preinstalled on Windows. I don't have Windows though, so I could definitely be wrong.

Reviewer: @iosenberg That looks good to me. I agree with @FiskFan1999 that it would be good to have it at the top, along with the git installation instructions. Because the video doesn't include installation instructions, I think it's good to have them at the top so that video-preferring people can just read the installation instructions then watch the video.

3. In your favorite web browser, navigate to http://127.0.0.1:8000/member.html?name={your-last-name}
4. You should see a page with all your information on it!
	- If you see an error message, try to fix the error in members/{your-last-name}.json

> 	- If you don't see anything, check the spelling, quotations, and commas in your entry in memberslist.json

Reviewer: If people start at the homepage and click through to their page, they can make sure the new `memberslist.json` works.

```suggestion
3. In your favorite web browser, navigate to <http://127.0.0.1:8000/>. Click the link to the members directory.
4. You should see a page that includes your name. Click on it.
	- If you don't see your name, check the spelling, quotations, and commas in your entry in memberslist.json
5. You should see a page with all your information on it!
	- If you see an error message, try to fix the error in members/{your-last-name}.json
```

Author: Ahh good point. Also just looks a lot nicer and is easier to navigate to.
